src/libpkcs11d.c

```diff
@@ -86,33 +86,49 @@ static int pkcs11d_rsa_private_common(const char *op, int flen, const unsigned c
         char buffer[4096];
         int l, slen = 0;
 
+        if(!b) {
+            fprintf(stderr, "Error creating BIO\n");
+        } else {
+            fprintf(stderr, "BIO created fine\n");
+        }
+
         BIO_set_conn_port(b, "1234");
         b = BIO_push(buf, b);
 
         /* Insert implementation here */
 
+        fprintf(stderr, "POST /%s/rsa/%s HTTP/1.0\n", op, pkd->id);
+        fprintf(stderr, "Content-Length: %d\n\n", flen);
```

Comment:
These are also more CWE-20s due to use of fprintf() inherent weakness to local side-channel attacks but probably some-what difficult to exploit. Cleaning this up is left as technical debt for now.